### ISSUE TITLE
Error reason detail missing for 422 errors

### ISSUE THREAD

**Issue by Xcodo** [zVolodymyr/Harvest.Api#7]
I had an issue yesterday where I was putting the wrong ID in a query (stupid me). It took me a while to debug what was going on, partly because the exception only showed the 422 status code name (`Unprocessable Entity`).

The Harvest API documentation has this line

Code | Explanation
-- | --
200 | Your request was successful.
201 | A new object has been created. Its representation will be returned in the response body.
403 | The object you requested was found but you don’t have authorization to perform your request.
404 | The object you requested can’t be found.
422 | There were errors processing your request. :star:Check the response body for additional information.:star:
429 | Your request has been throttled. Refer to the Rate Limiting section for details.
500 | There was a server error. Contact support@getharvest.com for help.

Would you include a pull request that adds the response body to the exception message in the case of a 422 error?

**Comment by zVolodymyr** [comment]
I have doubts about about just adding HttpResponse into common exception, because it means it should keep reponse not disposed. Maybe it will be better to create separate exception class for this error.

**Comment by steveschmitt** [comment]
I agree that it would be good to get the response content for these exceptions.  Maybe it could just be appended to the exception message.

**Comment by steveschmitt** [comment]
@Xcodo Here's what I did to get the response data.  First, create a subclass of HttpClientHandler:
```cs
    public class ExHttpClientHandler : HttpClientHandler
    {
        /// <summary> Capture the response body when response status is 422 </summary>
        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
        {
            var response = await base.SendAsync(request, cancellationToken);
            if (response.StatusCode == System.Net.HttpStatusCode.UnprocessableEntity)
            {
                var message = await response.Content.ReadAsStringAsync();
                throw new Exception($"{response.ReasonPhrase}: {message}");
            }
            return response;
        }
    }
```
Then use it in your HarvestClient: 
```cs
    var client = HarvestClient.FromAccessToken(AGENT, TOKEN, httpClientHandler: new ExHttpClientHandler());
```
When there's a 422 response, you'll get the body content in the exception message.
